xskillscore/core/comparative.py

```diff
@@ -0,0 +1,147 @@
+                * ``confidence``: Positive boundary for the random walk at significance
+                    ``alpha``.
+
+
+        Examples
+        --------
+        >>> f1 = xr.DataArray(np.random.normal(size=(30)),
+        ...      coords=[('time', np.arange(30))])
+        >>> f2 = xr.DataArray(np.random.normal(size=(30)),
+        ...      coords=[('time', np.arange(30))])
+        >>> o = xr.DataArray(np.random.normal(size=(30)),
+        ...      coords=[('time', np.arange(30))])
+        >>> st = sign_test(f1, f2, o, dim='time', metric='mae', orientation='negative')
+        >>> st.sel(results='sign_test').plot()
+        >>> st.sel(results='confidence').plot(c='gray')
+        >>> (-1*st.sel(results='confidence')).plot(c='gray')
+
+        References
+        ----------
+            * DelSole, T., & Tippett, M. K. (2016). Forecast Comparison Based on Random
+              Walks. Monthly Weather Review, 144(2), 615–626. doi: 10/f782pf
+    """
+
+    def _categorical_metric(observations, forecasts, dim):
+        """Returns True where forecasts exactly equals observations"""
+        return observations == forecasts
+
+    if metric is not None:
+        # make sure metric is a callable
+        if isinstance(metric, str):
+            if metric == 'categorical':
+                metric = _categorical_metric
+                if orientation != 'positive':
+                    warnings.warn(
+                        'Changing to "orientation=positive" for consistency with "metric=categorical"',
+                        UserWarning,
+                    )
+                orientation = 'positive'
+            else:
+                import xskillscore as xs
+
+                if hasattr(xs, metric):
+                    metric = getattr(xs, metric)
+                else:
+                    raise ValueError(
+                        f'xskillscore.metric could not be derived from {metric}.'
+                    )
+        elif not callable(metric):
+            raise ValueError(
+                f'metric needs to be a function/callable or None, found {type(metric)}'
+            )
+        if observation is not None:
+            # Compare the forecasts and observation using metric
+            metric_f1o = metric(forecast1, observation, dim=[])
+            metric_f2o = metric(forecast2, observation, dim=[])
+        else:
+            raise ValueError(
+                'observations must be provided when metric is provided', UserWarning
+            )
+
+    else:  # if metric=None, already evaluated
+        if observation is not None:
+            warnings.warn(
+                'Ignoring provided observation because no metric was provided',
+                UserWarning,
+            )
+        metric_f1o = forecast1
+        metric_f2o = forecast2
+
+    # Adjust for orientation of metric
+    if orientation == 'positive' and metric != _categorical_metric:
```

Comment:
I don't think it's correct to have the `and metric != _categorical_metric` here. We need to adjust for the positive orientation of `_categorical_metric`, which returns `True` (1) where the forecast matches the observations and `False` (0), otherwise.